debug_script.py

```diff
@@ -0,0 +1,16 @@
+import numpy as np
```

Comment:
I appreciate your idea of making a free variable where to put jit compiled classes. but still, we need to hard code the spec of classes which then leaves us with some boilerplate code.

``debug_script`` shows that datafit breaks down when its spec doesn't match the types of ``X`` and ``y`` (the inputs).
IMO, we should opt for a solution (still haven't thought about one yet) where it's up to the class to manage types-conflicts

Hence the final look should be something like

```python
datafit = Quadratic(dtype, **params)
```

Reply:
The specs of the classes are already hardcoded: https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/penalties/separable.py#L11

The spec can be updated dynamically using the trick we have to generate Quadratic32. The script is just an example to show that it can work.
I don't think it is necessary to pass dtype when instanciating `datafit`. It can be inferred from X at the time of `fit`.

Reply:
I don't really understand this example: it does not use the new code I propose. The same exact code would fail on main, right ?

Maybe I'm not clearly exposing the design I want to try. I will code the rest later today and share at that point.

Reply:
The example is to illustrate what problem this PR could solve

Reply:
Let's do things one at a time. First, avoid having jitted classes as attributes, in order to be usable in GridSearchCV. Then, as you suggest it's a good idea to update the spec dynamically depending on the type of X being passed to fit.